sql/core/src/main/scala/org/apache/spark/sql/execution/command/ddl.scala

```diff
@@ -432,6 +445,10 @@ case class AlterTableChangeColumnCommand(
     }.getOrElse(throw QueryCompilationErrors.cannotFindColumnError(name, schema.fieldNames))
   }
 
+  // Change the dataType of the column.
+  private def withNewType(column: StructField, dataType: DataType): StructField =
+    column.copy(dataType = dataType)
```

Comment:
nit: This looks more concise than calling `withNewType`, do we really need to create this function?

Reply:
This was suggested by @johanl-db, please check https://github.com/apache/spark/pull/46734#discussion_r1613401422.